multi_model_experiment/05_global_token_validation_experiment.py

```diff
@@ -0,0 +1,202 @@
+
+subsetted_tokens = pd.read_csv("output/subsetted_tokens.tsv", sep="\t")
+token_filter_list = subsetted_tokens.tok.tolist()
+subsetted_tokens.head()
+
+distance_files = list(
+    Path("output/combined_inter_intra_distances").rglob("saved_*_distance.tsv")
+)
+print(len(distance_files))
+
+year_distance_map = {
+    re.search(r"\d+", str(year_file)).group(0): (pd.read_csv(str(year_file), sep="\t"))
+    for year_file in tqdm.tqdm(distance_files)
+}
+
+full_token_set_df = pd.concat(
+    [
+        year_distance_map[year] >> ply.query(f"tok in {token_filter_list}")
+        for year in tqdm.tqdm(year_distance_map)
+    ]
+)
+print(full_token_set_df.shape)
+full_token_set_df.head()
+
+# ## Load individual Models for a couple years to see the change
+
+all_inter_year_models = list(Path("output/inter_models").rglob("*tsv"))
+
+regex_pattern = r"\d+_0-\d+_0"
+all_inter_models_df = pd.concat(
+    [
+        pd.read_csv(str(file), sep="\t")
+        >> ply.query(f"year_pair.str.contains({regex_pattern})")
+        for file in all_inter_year_models
+    ]
+)
+all_inter_models_df >> ply.slice_rows(15)
+
+middle_estimate = (
+    all_inter_models_df
+    >> ply.query("year_pair=='2010_0-2011_0'")
+    >> ply.pull("global_distance")
+).mean()
+middle_estimate
+
+regex_pattern = r"(\d+)_0-(\d+)_0"
+percent_diff_one_year = (
+    all_inter_models_df
+    >> ply_tdy.extract(
+        "year_pair", into=["year_1", "year_2"], regex=regex_pattern, convert=True
+    )
+    >> ply.group_by("year_1", "year_2")
+    >> ply.query("year_2-year_1==1")
+    >> ply.define(pct_diff="mean(global_distance)/middle_estimate - 1")
+    >> ply.select("year_1", "year_2", "pct_diff")
+    >> ply.ungroup()
+    >> ply.distinct()
+    >> ply_tdy.unite("year_pair", "year_1", "year_2")
+)
+percent_diff_one_year
+
+# # Does distance increase across the years?
+
+all_distance_df = (
+    full_token_set_df
+    >> ply.arrange("tok")
+    >> ply.group_by("year_2")
+    >> ply.define(
+        avg_global_distance="mean(original_global_distance)",
+        avg_global_distance_qst="mean(global_distance_qst)",
+        avg_ratio_metric="mean(ratio_metric)",
+    )
+    >> ply.select(
+        "avg_global_distance", "avg_global_distance_qst", "avg_ratio_metric", "year_1"
+    )
+    >> ply.distinct()
+    >> ply.ungroup()
+    >> ply.query("year_2-year_1==1")
+    >> ply_tdy.unite("year_pair", "year_1", "year_2")
+)
+all_distance_df >> ply.slice_rows(15)
+
+global_qst_comparison = all_distance_df >> ply_tdy.gather(
+    "metric",
+    "distance_value",
+    ply.select(
+        "avg_global_distance_qst",
+        "avg_global_distance",
+        "avg_ratio_metric",
+    ),
+)
+global_qst_comparison.year_pair = pd.Categorical(
+    global_qst_comparison.year_pair.tolist()
+)
+global_qst_comparison.head(10)
+
+middle_map = dict(
+    global_qst_comparison
+    >> ply.query("year_pair=='2010_2011'")
+    >> ply.pull(["metric", "distance_value"])
+)
+
+pct_diff_df = (
```

Comment:
What are these other distance metrics? Is the `avg_global_dist` inter distance without the correction for the intra variation?

Reply:
>What are these other distance metrics? Is the avg_global_dist inter distance without the correction for the intra variation?

I'll add a description for this plot. Basically, I was trying to figure out how to account for inter and intra year variation. I first went through using a variation of Qst (formula in 04_novel_distance_calculations notebook). I then attempted to construct a few other metrics (i.e. the other non-spike lines) and see if there is a large difference between them. (Hence this figure). Overall, the ratio metric makes the most sense for my other experiment (changepoint detection) and I ended up going with that.